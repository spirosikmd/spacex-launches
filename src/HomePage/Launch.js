import React from 'react';
import PropTypes from 'prop-types';
import { Link } from '@reach/router';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import LaunchDateTime from '../LaunchDateTime';
import LaunchStatus from '../LaunchStatus';

const styles = theme => ({
  launch: {
    display: 'flex',
    alignItems: 'center',
  },
  launchInfo: {
    flexGrow: 1,
  },
  connector: {
    minWidth: theme.spacing.unit * 2,
    border: `0.5px dashed ${theme.palette.secondary.dark}`,
    marginTop: '1px',
  },
});

const Launch = ({ classes, launch }) => {
  const hadMissionIds = launch.missionIds.length > 0;
  const hasContent = launch.details || hadMissionIds;

  return (
    <div className={classes.launch}>
      <LaunchStatus
        isSuccessful={launch.isSuccessful}
        isFailed={launch.isFailed}
        isUpcoming={launch.isUpcoming}
        isInProgress={launch.isInProgress}
      />
      <div className={classes.connector} />
      <Card className={classes.launchInfo}>
        <CardHeader
          title={launch.missionName}
          subheader={
            <LaunchDateTime
              utcDate={launch.utcDate}
              isTentative={launch.isTentative}
            />
          }
        />
        {hasContent && (
          <CardContent>
            {launch.details &&
              (hadMissionIds ? (
                <Typography component="p" gutterBottom>
                  {launch.details}
                </Typography>
              ) : (
                <Typography component="p">{launch.details}</Typography>
              ))}
            {hadMissionIds && (
              <Typography color="textSecondary" component="p">
                ID: {launch.missionIds.join(', ')}
              </Typography>
            )}
          </CardContent>
        )}
        {!launch.isTentative && (
          <CardActions>
            <Button
              size="small"
              color="primary"
              component={Link}
              to={`/launches/${launch.flightNumber}`}
            >
              explore
            </Button>
          </CardActions>
        )}
      </Card>
    </div>
  );
};

export const LaunchPropType = {
  isSuccessful: PropTypes.bool.isRequired,
  isFailed: PropTypes.bool.isRequired,
  isUpcoming: PropTypes.bool.isRequired,
  isInProgress: PropTypes.bool.isRequired,
  isTentative: PropTypes.bool.isRequired,
  missionName: PropTypes.string.isRequired,
  utcDate: PropTypes.object.isRequired,
  details: PropTypes.string,
  missionIds: PropTypes.array.isRequired,
  flightNumber: PropTypes.number.isRequired,
};

Launch.propTypes = {
  classes: PropTypes.object.isRequired,
  launch: PropTypes.shape(LaunchPropType).isRequired,
};

export default withStyles(styles)(React.memo(Launch));
