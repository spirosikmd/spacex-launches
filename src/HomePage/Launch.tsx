import React from 'react';
import PropTypes from 'prop-types';
import { Link } from '@reach/router';
import {
  withStyles,
  Theme,
  WithStyles,
  createStyles,
} from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Button, { ButtonProps } from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import LaunchDateTime from '../LaunchDateTime';
import LaunchStatus from '../LaunchStatus';
import LaunchMissionIds from '../LaunchMissionIds';
import { LaunchData } from '../api';

const styles = (theme: Theme) =>
  createStyles({
    launch: {
      display: 'flex',
      alignItems: 'center',
    },
    launchInfo: {
      flexGrow: 1,
    },
    connector: {
      minWidth: theme.spacing(2),
      border: `0.5px dashed ${theme.palette.secondary.dark}`,
      marginTop: '1px',
    },
  });

interface LinkButtonProps extends ButtonProps {
  to: string;
}

const LinkButton = (props: LinkButtonProps) => (
  <Button {...props} component={Link as any} />
);

interface LaunchProps extends WithStyles<typeof styles> {
  launch: LaunchData;
}

const Launch = ({ classes, launch }: LaunchProps) => {
  const hadMissionIds = launch.missionIds.length > 0;
  const hasContent = launch.details || hadMissionIds;

  return (
    <div className={classes.launch}>
      <LaunchStatus
        isSuccessful={launch.isSuccessful}
        isFailed={launch.isFailed}
        isUpcoming={launch.isUpcoming}
        isInProgress={launch.isInProgress}
      />
      <div className={classes.connector} />
      <Card className={classes.launchInfo}>
        <CardHeader
          title={launch.missionName}
          subheader={
            <LaunchDateTime
              utcDate={launch.utcDate}
              isTentative={launch.isTentative}
            />
          }
        />
        {hasContent && (
          <CardContent>
            {launch.details &&
              (hadMissionIds ? (
                <Typography component="p" gutterBottom>
                  {launch.details}
                </Typography>
              ) : (
                <Typography component="p">{launch.details}</Typography>
              ))}
            {hadMissionIds && (
              <LaunchMissionIds missionIds={launch.missionIds} />
            )}
          </CardContent>
        )}
        {!launch.isTentative && (
          <CardActions>
            <LinkButton
              size="small"
              color="primary"
              to={`/launches/${launch.flightNumber}`}
            >
              explore
            </LinkButton>
          </CardActions>
        )}
      </Card>
    </div>
  );
};

export const LaunchPropType = {
  isSuccessful: PropTypes.bool.isRequired,
  isFailed: PropTypes.bool.isRequired,
  isUpcoming: PropTypes.bool.isRequired,
  isInProgress: PropTypes.bool.isRequired,
  isTentative: PropTypes.bool.isRequired,
  missionName: PropTypes.string.isRequired,
  utcDate: PropTypes.object.isRequired,
  details: PropTypes.string,
  missionIds: PropTypes.array.isRequired,
  flightNumber: PropTypes.number.isRequired,
};

Launch.propTypes = {
  classes: PropTypes.object.isRequired,
  launch: PropTypes.shape(LaunchPropType).isRequired,
};

export default withStyles(styles)(React.memo(Launch));
