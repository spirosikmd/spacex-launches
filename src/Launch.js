import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { withStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import GroupWorkRoundedIcon from '@material-ui/icons/GroupWorkRounded';

const styles = theme => ({
  status: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: '50%',
    padding: theme.spacing.unit / 2,
    zIndex: '1',
  },
  statusIcon: {
    fontSize: 16,
  },
  upcoming: {
    color: theme.palette.grey['500'],
  },
  success: {
    color: theme.palette.success.main,
  },
  fail: {
    color: theme.palette.error.main,
  },
  launch: {
    display: 'flex',
    alignItems: 'center',
  },
  launchInfo: {
    padding: theme.spacing.unit * 2,
    flexGrow: 1,
  },
  connector: {
    minWidth: theme.spacing.unit * 2,
    border: `0.5px dashed ${theme.palette.secondary.dark}`,
    marginTop: '1px',
  },
});

export const Launch = ({ classes, launch }) => {
  const statusClassName = classnames(classes.status, {
    [classes.success]: launch.isSuccessful,
    [classes.fail]: launch.isFailed,
    [classes.upcoming]: launch.isUpcoming,
  });

  return (
    <div className={classes.launch}>
      <Paper className={statusClassName} elevation={1}>
        <GroupWorkRoundedIcon className={classes.statusIcon} />
      </Paper>
      <div className={classes.connector} />
      <Paper className={classes.launchInfo}>
        <Grid container spacing={8}>
          <Grid item xs={12}>
            <Typography>{launch.missionName}</Typography>
          </Grid>
          <Grid item xs={12}>
            <Typography color="textSecondary">
              {launch.utcDate.toLocaleString()}
            </Typography>
          </Grid>
          <Grid item xs={12}>
            <Typography>{launch.details}</Typography>
          </Grid>
        </Grid>
      </Paper>
    </div>
  );
};

export const LaunchPropType = {
  isSuccessful: PropTypes.bool.isRequired,
  isFailed: PropTypes.bool.isRequired,
  isUpcoming: PropTypes.bool.isRequired,
  isTentative: PropTypes.bool.isRequired,
  missionName: PropTypes.string.isRequired,
  utcDate: PropTypes.object.isRequired,
  details: PropTypes.string,
};

Launch.propTypes = {
  classes: PropTypes.object.isRequired,
  launch: PropTypes.shape(LaunchPropType).isRequired,
};

export default withStyles(styles)(React.memo(Launch));
