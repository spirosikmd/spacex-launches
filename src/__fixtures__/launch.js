export function createLaunchResponse(data) {
  return {
    flight_number: 1,
    mission_name: 'Test',
    launch_date_utc: '2019-01-01T18:16:00.000Z',
    launch_success: true,
    upcoming: false,
    is_tentative: false,
    details: 'Details',
    mission_id: ['id1'],
    links: {
      mission_patch: 'missionPatch',
    },
    ...data,
  };
}

export function createLaunch(data) {
  return {
    flightNumber: 1,
    missionName: 'Test',
    utcDate: new Date('2019-01-01'),
    isSuccessful: true,
    isFailed: false,
    isUpcoming: false,
    isInProgress: false,
    isTentative: false,
    details: 'Details',
    missionIds: ['id1'],
    missionPatch: 'missionPatch',
    ...data,
  };
}
