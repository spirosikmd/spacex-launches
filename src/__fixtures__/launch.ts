import { LaunchResponseData, LaunchData } from '../api';

export function createLaunchResponse(
  data: LaunchResponseData
): LaunchResponseData {
  return {
    flight_number: 1,
    mission_name: 'Test',
    launch_date_utc: '2019-01-01T18:16:00.000Z',
    launch_success: true,
    upcoming: false,
    is_tentative: false,
    details: 'Details',
    mission_id: ['id1'],
    links: {
      mission_patch: 'missionPatch',
    },
    ...data,
  };
}

export function createLaunch(data: LaunchData): LaunchData {
  return {
    flightNumber: 1,
    missionName: 'Test',
    utcDate: new Date('2019-01-01'),
    isSuccessful: true,
    isFailed: false,
    isUpcoming: false,
    isInProgress: false,
    isTentative: false,
    details: 'Details',
    missionIds: ['id1'],
    missionPatch: 'missionPatch',
    ...data,
  };
}
