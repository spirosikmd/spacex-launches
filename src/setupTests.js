import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import { unwrap } from '@material-ui/core/test-utils';

configure({ adapter: new Adapter() });

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

global.shallow = (Component, props = {}) => {
  const ComponentWithoutStyles = unwrap(Component);
  return shallow(<ComponentWithoutStyles {...props} />);
};

global.mount = (Component, props = {}) => {
  const ComponentWithoutStyles = unwrap(Component);
  return mount(<ComponentWithoutStyles {...props} />);
};

global.Date = jest.fn().mockImplementation(() => {
  return {
    toUTCString: jest.fn().mockReturnValue('Fri, 11 Jan 2019 15:31:00 GMT'),
    getDate: jest.fn().mockReturnValue(3),
    getMonth: jest.fn().mockReturnValue(10),
    getFullYear: jest.fn().mockReturnValue(2019),
  };
});

// https://github.com/airbnb/enzyme/issues/1875
jest.mock('react', () => {
  const r = jest.requireActual('react');
  return { ...r, memo: x => x };
});
