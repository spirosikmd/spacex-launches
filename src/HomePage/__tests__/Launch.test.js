import React from 'react';
import { render } from '../../setupTests';
import Launch from '../Launch';
import { createLaunch } from '../../__fixtures__/launch';

describe('Launch', () => {
  let props;

  beforeEach(() => {
    props = {
      classes: {
        launch: 'launch',
        launchInfo: 'launchInfo',
        connector: 'connector',
      },
      launch: createLaunch(),
    };
  });

  describe('when a launch is successful', () => {
    it('renders successful', () => {
      const { asFragment } = render(<Launch {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when a launch is failed', () => {
    it('renders failed', () => {
      props.launch.isFailed = true;
      props.launch.isSuccessful = false;
      const { asFragment } = render(<Launch {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when a launch is upcoming', () => {
    it('renders upcoming', () => {
      props.launch.isUpcoming = true;
      props.launch.isSuccessful = false;
      const { asFragment } = render(<Launch {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when launch is in progress', () => {
    it('renders in progress', () => {
      props.launch.isInProgress = true;
      props.launch.isSuccessful = false;
      const { asFragment } = render(<Launch {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when launch does not have details', () => {
    it('does not render the details section', () => {
      props.launch = createLaunch({ details: null });
      const { asFragment } = render(<Launch {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when launch does not have mission ids', () => {
    it('does not render the mission ids info', () => {
      props.launch = createLaunch({ missionIds: [] });
      const { asFragment } = render(<Launch {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when launch is tentative', () => {
    it('renders without actions', () => {
      props.launch = createLaunch({ isTentative: true });
      const { asFragment } = render(<Launch {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
