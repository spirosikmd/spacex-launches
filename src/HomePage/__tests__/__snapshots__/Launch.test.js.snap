// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch when a launch is failed renders failed 1`] = `
<DocumentFragment>
  <div
    class="Launch-launch-116 launch"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 LaunchStatus-status-119 LaunchStatus-fail-123 MuiPaper-rounded"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root LaunchStatus-statusIcon-120"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8 17.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM9.5 8c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8zm6.5 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
    </div>
    <div
      class="Launch-connector-118 connector"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root Launch-launchInfo-117 launchInfo MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Test
          </span>
          <span
            class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span>
              Tue, 01 Jan 2019 00:00:00 GMT
            </span>
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Details
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          ID: id1
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall"
          href="/launches/1"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            explore
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Launch when a launch is successful renders successful 1`] = `
<DocumentFragment>
  <div
    class="Launch-launch-1 launch"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 LaunchStatus-status-4 LaunchStatus-success-7 MuiPaper-rounded"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root LaunchStatus-statusIcon-5"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8 17.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM9.5 8c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8zm6.5 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
    </div>
    <div
      class="Launch-connector-3 connector"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root Launch-launchInfo-2 launchInfo MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Test
          </span>
          <span
            class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span>
              Tue, 01 Jan 2019 00:00:00 GMT
            </span>
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Details
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          ID: id1
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall"
          href="/launches/1"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            explore
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Launch when a launch is upcoming renders upcoming 1`] = `
<DocumentFragment>
  <div
    class="Launch-launch-231 launch"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 LaunchStatus-status-234 LaunchStatus-upcoming-236 MuiPaper-rounded"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root LaunchStatus-statusIcon-235"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8 17.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM9.5 8c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8zm6.5 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
    </div>
    <div
      class="Launch-connector-233 connector"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root Launch-launchInfo-232 launchInfo MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Test
          </span>
          <span
            class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span>
              Tue, 01 Jan 2019 00:00:00 GMT
            </span>
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Details
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          ID: id1
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall"
          href="/launches/1"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            explore
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Launch when launch does not have details does not render the details section 1`] = `
<DocumentFragment>
  <div
    class="Launch-launch-461 launch"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 LaunchStatus-status-464 LaunchStatus-success-467 MuiPaper-rounded"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root LaunchStatus-statusIcon-465"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8 17.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM9.5 8c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8zm6.5 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
    </div>
    <div
      class="Launch-connector-463 connector"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root Launch-launchInfo-462 launchInfo MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Test
          </span>
          <span
            class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span>
              Tue, 01 Jan 2019 00:00:00 GMT
            </span>
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          ID: id1
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall"
          href="/launches/1"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            explore
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Launch when launch does not have mission ids does not render the mission ids info 1`] = `
<DocumentFragment>
  <div
    class="Launch-launch-576 launch"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 LaunchStatus-status-579 LaunchStatus-success-582 MuiPaper-rounded"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root LaunchStatus-statusIcon-580"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8 17.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM9.5 8c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8zm6.5 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
    </div>
    <div
      class="Launch-connector-578 connector"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root Launch-launchInfo-577 launchInfo MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Test
          </span>
          <span
            class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span>
              Tue, 01 Jan 2019 00:00:00 GMT
            </span>
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Details
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall"
          href="/launches/1"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            explore
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Launch when launch is in progress renders in progress 1`] = `
<DocumentFragment>
  <div
    class="Launch-launch-346 launch"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 LaunchStatus-status-349 LaunchStatus-inProgress-354 MuiPaper-rounded"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root LaunchStatus-statusIcon-350"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </div>
    <div
      class="Launch-connector-348 connector"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root Launch-launchInfo-347 launchInfo MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Test
          </span>
          <span
            class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span>
              Tue, 01 Jan 2019 00:00:00 GMT
            </span>
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Details
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          ID: id1
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall"
          href="/launches/1"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            explore
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Launch when launch is tentative renders without actions 1`] = `
<DocumentFragment>
  <div
    class="Launch-launch-691 launch"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 LaunchStatus-status-694 LaunchStatus-success-697 MuiPaper-rounded"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root LaunchStatus-statusIcon-695"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8 17.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM9.5 8c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8zm6.5 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
    </div>
    <div
      class="Launch-connector-693 connector"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root Launch-launchInfo-692 launchInfo MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Test
          </span>
          <span
            class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span>
              Tue, 01 Jan 2019
            </span>
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Details
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          ID: id1
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
